src/io/linux.zig

```diff
@@ -503,6 +738,7 @@ pub const IO = struct {
                                 .OPNOTSUPP => error.OperationNotSupported,
                                 .PIPE => error.BrokenPipe,
                                 .TIMEDOUT => error.ConnectionTimedOut,
+                                .CANCELED => error.Canceled,
```

Comment:
Wait, shouldn't this _remain_ unexpected? I think we are guaranteed to _not_ complete canceled completions? 

Ah, and there's also a clear bug here: we _don't_ add `Canceled` to `pub const ReadError = error{` and friends.  That is, what happens here is that we essentially downcast an `anyerror` to a well-typed error (by casting the function pointer), and that would be an invalid cast. 

🤔 


perhaps we should't type `const result: anyerror!usize` and rather use `const result: SendError!usize` here?

Reply:
> Wait, shouldn't this remain unexpected? I think we are guaranteed to not complete canceled completions?

Good point!

> perhaps we should't type const result: anyerror!usize and rather use const result: SendError!usize here?

This found a bug! `CancelError` was missing ` || posix.UnexpectedError`.